packages/pds/src/app-view/services/indexing/processor.ts

```diff
@@ -180,7 +174,29 @@ export class RecordProcessor<T, S> {
         record,
         found.indexedAt,
       )
-      return this.params.eventsForDelete(deleted, inserted)
+      await this.handleNotifs({ deleted, inserted: inserted ?? undefined })
+    }
+  }
+
+  async handleNotifs(op: { deleted?: S; inserted?: S }) {
+    let notifs: UserNotification[] = []
+    if (op.deleted) {
+      const forDelete = this.params.notifsForDelete(
+        op.deleted,
+        op.inserted ?? null,
+      )
+      if (forDelete.toDelete) {
```

Comment:
May just protect us from ourselves, when we eventually return a `toDelete: []` somewhere.
```suggestion
      if (forDelete.toDelete?.length) {
```